### PR TITLE
Add popularity datasets

### PR DESCRIPTION
**Pull request by amCap1712**
Add job to import and process MLHD+ dumps (likely this will change in future when I again get back to efficiently processing the dump).

Add jobs to generate popularity data from the MLHD+ dump.

### CODE REVIEW

File: listenbrainz_spark/mlhd/popularity.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+from listenbrainz_spark import config
+from listenbrainz_spark.path import MLHD_PLUS_DATA_DIRECTORY
+from listenbrainz_spark.stats import run_query
+
+
+STATS_PER_MESSAGE = 10000
+
+
+def generate(name, query):
```
Comment (mayhem) [inline listenbrainz_spark/mlhd/popularity.py:11]: this name is a bit too generic. generate_popularity_stats?

File: listenbrainz_spark/mlhd/similarity.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+from listenbrainz_spark.path import RECORDING_LENGTH_DATAFRAME, MLHD_PLUS_DATA_DIRECTORY
+from listenbrainz_spark.stats import run_query
+from listenbrainz_spark.utils import read_files_from_HDFS
+
+
+DEFAULT_TRACK_LENGTH = 180
+
+
+def build_partial_sessioned_index(listen_table, metadata_table, session, max_contribution, skip_threshold):
+    # TODO: Handle case of unmatched recordings breaking sessions!
```
Comment (mayhem) [inline listenbrainz_spark/mlhd/similarity.py:13]: Remove, implement, add ticket?

**Review by mayhem — APPROVED** [review]
Part how fast can you type, part very meaty. Nice work!

File: listenbrainz_spark/mlhd/popularity.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz_spark/mlhd/popularity.py:11]: fixed

File: listenbrainz_spark/mlhd/similarity.py (hunk 2 above)
Comment (amCap1712) [reply listenbrainz_spark/mlhd/similarity.py:13]: removed.

**Pull request merged**